strict-mvar/test/Test/Control/Concurrent/Class/MonadMVar/Strict/WHNF.hs

```diff
@@ -0,0 +1,201 @@
+{-# LANGUAGE CPP              #-}
+{-# LANGUAGE RankNTypes       #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+{-# HLINT ignore "Use camelCase" #-}
+
+module Test.Control.Concurrent.Class.MonadMVar.Strict.WHNF (tests) where
+
+#if CHECKED
+import           Control.Concurrent.Class.MonadMVar.Strict.Checked
+#else
+import           Control.Concurrent.Class.MonadMVar.Strict
+#endif
+import           Control.Monad (void)
+import           Data.Typeable (Proxy (..), Typeable)
+import           NoThunks.Class (OnlyCheckWhnf (OnlyCheckWhnf), unsafeNoThunks)
+import           Test.QuickCheck.Monadic (PropertyM, monadicIO, monitor, run)
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.QuickCheck (Fun, applyFun, counterexample,
+                     testProperty)
+import           Test.Utils (monadicSim)
+
+{-------------------------------------------------------------------------------
+  Main test tree
+-------------------------------------------------------------------------------}
+
+name :: String
+#if CHECKED
+name = "Strict.Checked"
+#else
+name = "Strict"
+#endif
+
+tests :: TestTree
+tests = testGroup ("Test.Control.Concurrent.Class.MonadMVar." <> name) [
+      testGroup "WHNF" [
+          testGroup "IO" [
+              testProperty "prop_newMVar" $
+                monadicIO .: prop_newMVar (Proxy @Int)
```

Comment:
Is it worth to keep the tests, e.g. `prop_newMVar` polymorphic if we only instantiate them with `Int`?  I think it would be cleaner to monomorphise them.